examples/aggregator/module/rust/src/router.rs

```diff
@@ -0,0 +1,137 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::{
+    handler::Handler,
+    proto::oak::examples::aggregator::{AggregatorDispatcher, HandlerInit},
+    Config,
+};
+use anyhow::Context;
+use log::debug;
+use oak::{
+    io::{ReceiverExt, Sender, SenderExt},
+    proto::oak::{
+        application::ConfigMap,
+        invocation::{GrpcInvocation, GrpcInvocationSender},
+    },
+};
+use oak_abi::label::{confidentiality_label, web_assembly_module_tag, Label};
+use oak_services::proto::google::rpc;
+
+/// A node that routes each incoming gRPC invocation to a Handler Node that can handle requests with
+/// the label of the incoming request.
+///
+/// This node never looks at the contents of the invocation messages, only at the labels of its
+/// channels, and therefore keeps a public confidentiality label, which also allows it to create
+/// further nodes and channels, with more specific labels.
+#[derive(Default)]
+pub struct Router {
+    /// Invocation sender channel half for Handler Node.
+    handler_command_sender: Sender<oak::grpc::Invocation>,
+    /// Confidentiality label corresponding to Aggregator WebAssembly module SHA-256 hash.
+    aggregator_hash_label: Label,
+}
+
+impl oak::CommandHandler for Router {
+    type Command = oak::grpc::Invocation;
+
+    fn handle_command(&mut self, command: oak::grpc::Invocation) -> anyhow::Result<()> {
+        // The router node has a public confidentiality label, and therefore cannot read the
+        // contents of the request of the invocation (unless it happens to be public as well), but
+        // it can always inspect its label.
+        match (&command.receiver, &command.sender) {
+            (Some(receiver), Some(sender)) => {
+                let label = receiver.label()?;
+                let grpc_response_writer = oak::grpc::ChannelResponseWriter::new(sender.clone());
+                // Check if request label is valid in the context of Aggregator application.
+                if label != self.aggregator_hash_label {
+                    grpc_response_writer
+                        .close(Err(oak::grpc::build_status(
+                            rpc::Code::InvalidArgument,
+                            "Invalid request label",
+                        )))
+                        .context("Couldn't send error response back")?;
+                    anyhow::bail!("Invalid request label: {:?}", label);
+                }
+
+                // Route gRPC invocations to Handler Node.
+                self.handler_command_sender
+                    .send(&command)
+                    .context("Couldn't send invocation to worker node")
+            }
+            _ => {
+                anyhow::bail!("Received malformed gRPC invocation");
+            }
+        }
+    }
+}
```

Comment:
I think, specifically for this it could be a wrapper around Channel, that checks privileges (probably does the same thing as Channel in Runtime), but does this in SDK.
Since having SDK (non-Runtime) errors was the only reason behind having this piece of code.